### PR TITLE
feat(mobile): searchable, sortable logbook (shared logic + amber filter)

### PR DESCRIPTION
**Pull request by azuttre**
## Summary

Brings the searchable / sortable logbook to the **mobile** app (the web logbook already had it), and shares the logic so both platforms stay in sync.

- **Mobile You → Logbook** gets a top toolbar: a climb-name search + an **amber** filter button (with an active-filter badge). The amber accent makes it obvious you're in logbook mode, distinct from the violet climb search.
- **Filter sheet** reuses the climb-filter primitives, with a **Latest / Hardest** sort preset at the very top, then **Refine** (status, flash-only, grade, angle) and **Advanced** (date range, benchmarks). One tap on a preset sorts without touching anything else.
- **"Hardest"** sorts by consensus grade → your *effective* grade (the grade you logged, or the consensus when you didn't grade it) → most recent ascent. This is a change to the shared `userAscentsFeed` preset, so **web's "Hardest" updates to match** too.
- New shared **`@boardsesh/logbook`** package holds the filter/sort/preset state, sanitizers, and the feed-input builder; **web was refactored to consume it** (net −156 lines) so the two platforms evolve together.

## What's where

- `packages/shared/logbook` — new pure-TS package (types, defaults, sanitizers, `toAscentFeedInput`) + 12 unit tests.
- `packages/backend/.../resolvers/ticks/queries.ts` — `hardest` preset = consensus → effective → date, with an integration test.
- `packages/mobile/.../components/you/` — `LogbookTab` toolbar, `LogbookFilterSheet`, `useLogbookSearch`.
- `packages/web/.../logbook-preferences.ts` + `logbook-feed.tsx` — refactored onto the shared package (behavior-preserving).
- i18n: `you` namespace keys in en-US / es / fr.

A follow-up commit on this branch adds logbook filter/sort **persistence** (expo-secure-store) before this is marked ready.

## Testing

- `typecheck` (web / mobile / backend / shared) clean; `check:mobile-bundle` OK.
- Shared: 12 unit tests (input builder + sanitizers). Backend: hardest-ordering integration test. Web: 239 logbook tests pass. Mobile: toolbar/sheet tests.

## Release Notes

Your logbook is searchable again on mobile. Head to You → Logbook to find a climb by name, narrow by grade, angle, date, or sends/attempts, and sort by **Latest** or **Hardest** so your hardest ticks rise to the top. The filter button is amber, so you always know you're searching your logbook, not the whole board.

### CODE REVIEW

**Comment by azuttre** [comment]
Thanks. Addressed in `d2c3429`:

**Fixed**
- **`sanitizeDate` validation:** now requires `YYYY-MM-DD` and rejects impossible dates, so a corrupted persisted value can't reach the backend date comparison (`packages/shared/logbook/src/sanitize.ts`).
- **`maximumDate` rebuilt each render:** the To-date row now uses a memoized `today`, so `DateRangeRow`'s `openAndroid` callback no longer re-arms every render (`LogbookFilterSheet.tsx`).

**Test gaps filled**
- `LogbookFilterSheet` component test: preset selection, flash auto-clear on attempts-only, Reset clears the search + restores defaults, and date reveal-without-commit.
- Hydration-gate test: the feed stays disabled until persisted prefs load, then opens (the guard against a default-then-persisted double fetch).
- `countActiveLogbookFilters` edge cases, including the angle range counted once.

**Already fixed in earlier commits on this branch**
- Angle badge double-count: now counts the angle range as one (`use-logbook-search.ts`).
- iOS "From" date committing today on tap: now reveals the picker without committing (`revealed` state in `DateRangeRow`).

**Design decision:** the filter badge intentionally reflects filters only, not the sort preset. Presets (Latest/Hardest) change ordering, not the result set, so they don't count as an active filter.

**Comment by azuttre** [comment]
Thanks for the pass. Folded in the two trivial ones on `4a2bcd6` and held the rest as intentional:

**Folded in**
- `SearchHeader` `onFocus`/`onBlur` are now optional, so the logbook toolbar drops its noop passthroughs (the climbs screen still passes its real handlers).
- Added a comment on the sheet-mount `today` ceiling noting the midnight-rollover tradeoff.

**Held (intentional / out of scope)**
- `sanitizeDate` strict ISO validation — this is the validation an earlier review asked for. The date picker only emits `YYYY-MM-DD`, and a non-ISO persisted value was already invalid (it would have hit the backend date comparison), so dropping it is the correct behavior, not a silent regression.
- `GRADE_SCALE_BOARD = 'kilter'` — Kilter and Tension share the difficulty-id scale, and the logbook spans boards so there's no single "active board" to key off. MoonBoard-aware grade filtering in the logbook is a separate enhancement, not part of this PR.
- `JSON.stringify(feedInput)` query key — `toAscentFeedInput` builds the object deterministically, so the key is stable in practice. Low risk as flagged.

CI is green on `4a2bcd6`; this is just waiting on the OTA-environment approval.

**Comment by azuttre** [comment]
hey, the /pr-preview didn't kick anything off — I think the command is actually /ota-preview. But the real blocker is the approval gate: the build for PR 3179 is waiting on the "ota-preview" environment.
Can you open https://github.com/boardsesh/boardsesh/actions/runs/28126229277 and hit "Review deployments → Approve and deploy"? Once that publish job goes green I should be able to switch to the pr-3179 channel in the app and test. thanks!

**Comment by azuttre** [comment]
Thanks for approving the publish @marcodejongh — `pr-3179` is live on both platforms.

One blocker on my end: the OTA Channel Switcher doesn't show up under More. It's gated on the tester role (`userIsTester` looks for a `tester`/`admin` row in `community_roles`), and my account doesn't have one, so the whole Development section stays hidden. The feature itself is in the 2.0.1 build, so nothing's missing client-side, it's just the role.

`grantRole` is admin-only, so could you grant me `tester` (global is fine) via the admin role screen, or `grantRole({ userId: <mine>, role: tester, boardType: null })`? Once it's on I'll re-login to refresh my profile and switch to `pr-3179` to test.

**Comment by azuttre** [comment]
For the grant: my Boardsesh username is **Helter_Skilter**

File: packages/web/package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "@boardsesh/graphql": "*",
     "@boardsesh/graphql-client": "*",
     "@boardsesh/i18n": "*",
+    "@boardsesh/logbook": "*",
```
[the remark this answers is not in the document]
Comment (azuttre) [reply packages/web/package.json:47]: Good catch, confirmed and fixed in `9b3a47f`. Added `@boardsesh/logbook` to `transpilePackages` in `packages/web/next.config.mjs`, alongside the other internal TS workspaces (the package exports `src/index.ts` directly). Our `vp` validation runs typecheck rather than `next build`, so this wouldn't have shown up until a production web build.

**Comment by azuttre** [comment]
Thanks, this was a good pass — addressed all of it in `4c1a24d`.

**Correctness**
- **First-launch race** — added a `userModified` flag to the reducer. A filter/sort change applied before `loadLogbookPrefs` resolves is kept (the late `hydrate` only opens the gate instead of overwriting), and the persistence effect then saves it. Covered by a new hook test that applies before resolving the load.
- **`saveLogbookPrefs` unhandled rejection** — now wraps the write in try/catch and swallows, mirroring `loadLogbookPrefs`.
- **Inverted grade range** — `sanitizeLogbookFilters` now swaps `minGrade`/`maxGrade` when inverted. (The angle range was already clamped to `max >= min`, so only grade needed it.) New sanitize test.

**Performance**
- **Angle `Chip`** — now `memo`'d and takes a `value` with a stable `onPress(value)`, so the rails pass `handleMinAngle`/`handleMaxAngle` directly instead of a per-chip arrow. Unrelated filter changes no longer re-render the ~30 chips.

**Backend validation**
- `fromDate`/`toDate` in `AscentFeedInputSchema` now require `YYYY-MM-DD`, so a direct GraphQL call can't bypass the client sanitizer.

**Tests**
- `useLogbookSearch`: hydration-on-mount, persistence-on-change, the apply-before-hydrate race, and reset.
- Hardest sort: added a case where consensus + effective grade match and only `climbedAt` differs, asserting the more-recent ascent sorts first (exercises the third ORDER BY key).

Validation: typecheck (shared/mobile/backend), lint, fmt all clean; mobile logbook 40 / shared 14 / backend tick-queries 33 pass.

**Comment by azuttre** [comment]
heads up @marcodejongh: CI is red on this PR but I don't think it's the feature. typecheck, the Metro bundle, and tests all pass on a clean local install, and it runs fine on the pr-3179 build.

The failures are module-resolution errors in CI's `vp run typecheck` for the one new file (LogbookFilterSheet.tsx): it can't resolve `@gorhom/bottom-sheet` and two sibling files, while the 25+ other files importing `@gorhom` pass. They're unchanged so they're cache-skipped; my file is "affected" so it gets a cold re-run and trips. Doesn't reproduce locally on any clean install, and main's typecheck only stays green because the mobile package is a cache hit. Looks like the same CI tooling issue that keeps main red, not the code.

No rush to merge though. I'd rather get the OTA preview working and test it first, which just needs the tester role on my account (Helter_Skilter) whenever you get a chance.

**Comment by marcodejongh** [comment]
@azuttre Added you as tester, but I think the fignerprint of the app moved on, so you cant load the OTA (did give tester). Just test the sideload or the ios test flight.

**Comment by marcodejongh** [comment]
@azuttre The build failures were legit, it was because I removed gorhom sheet on main, and replaced it with a native thing. But I'll swap it over now
